### PR TITLE
Concat doc

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Moved concat in the list of completed ops in the `sharing work` page, and added a page on steps to add and test an operation

### CODE REVIEW

File: docs/HowToAddAnOperation.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# How to support to process a new ONNX Operation
+
+The text below describe the list of actions needed to successfully process a new ONNX operation. We are adding here support for an existing operation, thus when parsing the ONNX list of operation form the ONNX repository, this operation is already present. The step below explain how to add the necessary support to process that operation and lower it to code that can be executed using the LLVM compiler backend.
+
+In the example below, we assume that we add support for the Concat ONNX operation. All paths are relatives to the root directory of onnx-mlir distribution.
```
Comment (tjingrant) [inline docs/HowToAddAnOperation.md:5]: onnx-mlir distribution -> onnx-mlir repository?

File: docs/HowToAddAnOperation.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+## Generate the proper ONNX.td.inc
+
+The first step is to add support so that MLIR can determine the output type and shape from its input variables and parameters. This step is called “Shape Inference.” The first step is to indicate in the automatically generated `ONNXOps.td.inc` that the new operation needs to invoke shape inference.
```
Comment (tjingrant) [inline docs/HowToAddAnOperation.md:10]: needs to invoke shape inference -> implements the shape inference interface method?

File: docs/HowToAddAnOperation.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Lowering to krnl dialect
+
+Files related to the lowering of the new operations resides in the `src/Conversion/ONNXtoKRNL` directory and subdirectories. For the `concat` operation, we added code to lower it to krnl dialect in the ` src/Conversion/ONNXToKrnl/Tensor/concat.cpp` file. See other similar lowering for inspiration. We suggest to use `assert` statements for any unexpected values encountered while lowering the operation, as illegal parameter values should be caught in the shape inference phase, not successive passes such as lowering to the krnl dialect.
+
+In that file, the `populateLoweringONNXConcatOpPattern` operation (where `Concat` would be replaced with the actual new operation) will need to be defined in ` src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp` and invoked in the ` runOnModule` function in the ` src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp` file.
+
+To compile properly, you will also need to add the new `.cpp` file in the ` src/Conversion/ONNXToKrnl/cMakeLists.txt` file.
```
Comment (tjingrant) [inline docs/HowToAddAnOperation.md:49]: cMakeLists.txt -> CMakeLists.txt

File: docs/HowToAddAnOperation.md (hunk 1 above)
Comment (AlexandreEichenberger) [reply docs/HowToAddAnOperation.md:5]: tx

File: docs/HowToAddAnOperation.md (hunk 3 above)
Comment (AlexandreEichenberger) [reply docs/HowToAddAnOperation.md:49]: tx

File: docs/HowToAddAnOperation.md (hunk 2 above)
Comment (AlexandreEichenberger) [reply docs/HowToAddAnOperation.md:10]: yes, tx

**Comment by AlexandreEichenberger** [comment]
thanks for suggestions

**Pull request merged**